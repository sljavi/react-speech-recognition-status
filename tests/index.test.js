import React from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import SpeechRecognitionStatus from './../src/index';
import _ from 'lodash';

const defaultProps = {
  contextName: '',
  error: '',
  connecting: false,
  recording: true,
  final: false,
  text: 'foo',
  sleeping: false
};

function render(props = {}) {
  const updatedProps = {...defaultProps, ...props};
  return renderer
    .create(<SpeechRecognitionStatus {...updatedProps}/>)
    .toJSON();
}

describe('speech recognition status component', () => {
  it('should render without throwing an error', function() {
    const el = render();
    expect(el).toMatchSnapshot();
  });

  it('should render the context name when is set', function() {
    const el = render({
      contextName: 'bar'
    });
    expect(el).toMatchSnapshot();
  });

  it('should render add the class final to the text speech', function() {
    const el = render({
      error: '',
      connecting: false,
      recording: true,
      final: true,
    });

    expect(el).toMatchSnapshot();
  });

  it('should render connecting icon', function() {
    const el = render({
      error: '',
      connecting: true,
      recording: false,
      final: false
    });

    expect(el).toMatchSnapshot();
  });

  it('should render error icon', function() {
    const el = render({
      error: 'some error',
      connecting: false,
      recording: true,
      final: false
    });

    expect(el).toMatchSnapshot();
  });

  it('should render recording icon', function() {
    const el = render({
      error: '',
      connecting: false,
      recording: true,
      final: false
    });

    expect(el).toMatchSnapshot();
  });

  it('should render recording icon with volume level', function() {
    const el = render({
      error: '',
      connecting: false,
      recording: true,
      final: false,
      volume: 70
    });

    expect(el).toMatchSnapshot();
  });

  it('should render mute icon', function() {
    const el = render({
      error: '',
      connecting: false,
      recording: false,
      final: false
    });

    expect(el).toMatchSnapshot();
  });

  it('should render compact view', function() {
    const el = render({
      compact: true
    });

    expect(el).toMatchSnapshot();
  });

  it('should render sleeping compact view', function() {
    const el = render({
      compact: true,
      sleeping: true
    });

    expect(el).toMatchSnapshot();
  });

});
