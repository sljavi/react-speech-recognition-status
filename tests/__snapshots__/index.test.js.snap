// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`speech recognition status component should render add the class final to the text speech 1`] = `
<div
  className="speech-recognition-status expanded"
>
  <div
    className="status"
  >
    <div
      className="speech"
    >
      <span
        className="current-speech final"
      >
        foo
      </span>
    </div>
    <span
      className="status-icon-container"
    >
      <i
        className="fa fa-fw fa-microphone"
      />
    </span>
  </div>
</div>
`;

exports[`speech recognition status component should render compact view 1`] = `
<div
  className="speech-recognition-status compact"
>
  <span
    className="status-icon-container"
  >
    <i
      className="fa fa-fw fa-microphone"
    />
  </span>
  <span
    className="status-details"
  >
    <span
      className="current-speech"
    >
      foo
    </span>
  </span>
</div>
`;

exports[`speech recognition status component should render connecting icon 1`] = `
<div
  className="speech-recognition-status expanded"
>
  <div
    className="status"
  >
    <div
      className="speech"
    >
      <span
        className="current-speech"
      >
        foo
      </span>
    </div>
    <span
      className="status-icon-container"
    >
      <i
        className="fa fa-fw fa-refresh fa-spin"
      />
    </span>
  </div>
</div>
`;

exports[`speech recognition status component should render error icon 1`] = `
<div
  className="speech-recognition-status expanded"
>
  <div
    className="status"
  >
    <div
      className="speech"
    >
      <span
        className="current-speech"
      >
        foo
      </span>
    </div>
    <span
      className="status-icon-container"
    >
      <i
        className="fa fa-fw fa-exclamation-circle fa-microphone"
      />
    </span>
  </div>
</div>
`;

exports[`speech recognition status component should render mute icon 1`] = `
<div
  className="speech-recognition-status expanded"
>
  <div
    className="status"
  >
    <div
      className="speech"
    >
      <span
        className="current-speech"
      >
        foo
      </span>
    </div>
    <span
      className="status-icon-container"
    >
      <i
        className="fa fa-fw fa-microphone-slash"
      />
    </span>
  </div>
</div>
`;

exports[`speech recognition status component should render recording icon 1`] = `
<div
  className="speech-recognition-status expanded"
>
  <div
    className="status"
  >
    <div
      className="speech"
    >
      <span
        className="current-speech"
      >
        foo
      </span>
    </div>
    <span
      className="status-icon-container"
    >
      <i
        className="fa fa-fw fa-microphone"
      />
    </span>
  </div>
</div>
`;

exports[`speech recognition status component should render the context name when is set 1`] = `
<div
  className="speech-recognition-status expanded"
>
  <div
    className="status"
  >
    <div
      className="speech"
    >
      <span
        className="context"
      >
        <span>
          bar
        </span>
      </span>
      <span
        className="current-speech"
      >
        foo
      </span>
    </div>
    <span
      className="status-icon-container"
    >
      <i
        className="fa fa-fw fa-microphone"
      />
    </span>
  </div>
</div>
`;

exports[`speech recognition status component should render without throwing an error 1`] = `
<div
  className="speech-recognition-status expanded"
>
  <div
    className="status"
  >
    <div
      className="speech"
    >
      <span
        className="current-speech"
      >
        foo
      </span>
    </div>
    <span
      className="status-icon-container"
    >
      <i
        className="fa fa-fw fa-microphone"
      />
    </span>
  </div>
</div>
`;
